prusti-viper/src/verifier.rs

```diff
@@ -74,204 +86,318 @@ impl<'v, 'tcx> Verifier<'v, 'tcx> {
+                            Some(span) => {
+                                let key = (pos_id, program_name.clone());
+                                if !quantifier_instantiations.contains_key(&key) {
+                                    quantifier_instantiations.insert(key.clone(), FxHashMap::default());
+                                }
+                                let map = quantifier_instantiations.get_mut(&key).unwrap();
+                                // this replaces the old entry which is exactly what we want
+                                map.insert(q_name, insts);
+                                let mut n: u64 = 0;
+                                for insts in map.values() {
+                                    n += *insts;
+                                }
+                                PrustiError::message(
+                                    format!("quantifierInstantiationsMessage{}",
+                                        json!({"instantiations": n, "method": program_name}),
+                                    ), span.clone()
+                                ).emit(&self.env.diagnostic);
+                            },
+                            None => error!("#{insts} quantifier instantiations of {q_name} for unknown position id {pos_id} in verification of {program_name}"),
+                        }
+                    }
+                }
+                ServerMessage::QuantifierChosenTriggers {
+                    viper_quant,
+                    triggers,
+                    pos_id,
+                } => {
+                    if config::report_viper_messages() && pos_id != 0 {
+                        match self.encoder.error_manager().position_manager().get_span_from_id(pos_id) {
+                            Some(span) => {
+                                PrustiError::message(
+                                    format!("quantifierChosenTriggersMessage{}",
+                                        json!({"viper_quant": viper_quant, "triggers": triggers}),
+                                    ), span.clone()
+                                ).emit(&self.env.diagnostic);
+                            },
+                            None => error!("Invalid position id {pos_id} for viper quantifier {viper_quant} in verification of {program_name}"),
+                        }
                     }
                 }
             }
-
-            prusti_errors.push(prusti_error);
         }
-        prusti_errors.sort();
 
-        for prusti_error in prusti_errors {
-            debug!("Prusti error: {:?}", prusti_error);
-            if prusti_error.is_disabled() {
-                prusti_error.cancel();
-            } else {
+        // if we are in an ide, we already emit the errors asynchronously, otherwise we wait for
+        // all of them in order to sort them
+        if !config::show_ide_info() {
+            prusti_errors.sort();
```

Comment:
The comment about sortedness (just before `let mut prusti_errors`) should be here.